### ISSUE TITLE
I can't run the app after installing the package

### ISSUE THREAD

**Issue by luisms89** [MaikuB/flutter_local_notifications#254]
HI I'm trying to install the last version of the package but when I try to run the app it throws the following error:

```
Execution failed for task ':app:processDebugResources'.
> Android resource linking failed
  Output:  <...>\build\app\intermediates\incremental\mergeDebugResources\merged.dir\values\values.xml:26: error: resource android:attr/fontVariationSettings not found.
  <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\incremental\mergeDebugResources\merged.dir\values\values.xml:26: error: resource android:attr/ttcIndex not found.
  error: failed linking references.

  Command: <...>\.gradle\caches\transforms-1\files-1.1\aapt2-3.2.1-4818971-windows.jar\8c2ae7748f4c761862eef3bf251d07ee\aapt2-3.2.1-4818971-windows\aapt2.exe link -I\
         <...>\AppData\Local\Android\sdk\platforms\android-27\android.jar\
          --manifest\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\merged_manifests\debug\processDebugManifest\merged\AndroidManifest.xml\
          -o\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\processed_res\debug\processDebugResources\out\resources-debug.ap_\
          -R\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\incremental\processDebugResources\resources-list-for-resources-debug.ap_.txt\
          --auto-add-overlay\
          --java\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\generated\not_namespaced_r_class_sources\debug\processDebugResources\r\
          --custom-package\
          com.example.ntcdapp\
          -0\
          apk\
          --output-text-symbols\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\symbols\debug\R.txt\
          --no-version-vectors
  Daemon:  AAPT2 aapt2-3.2.1-4818971-windows Daemon #0
Output:  <...>\.gradle\caches\transforms-1\files-1.1\core-1.0.1.aar\51299e896ee984661bdb619accdf31e6\res\values\values.xml:89:5-125:25: AAPT: error: resource android:attr/fontVariationSettings not found.

  <...>\.gradle\caches\transforms-1\files-1.1\core-1.0.1.aar\51299e896ee984661bdb619accdf31e6\res\values\values.xml:89:5-125:25: AAPT: error: resource android:attr/ttcIndex not found.

  error: failed linking references.
  Command: <...>\.gradle\caches\transforms-1\files-1.1\aapt2-3.2.1-4818971-windows.jar\8c2ae7748f4c761862eef3bf251d07ee\aapt2-3.2.1-4818971-windows\aapt2.exe link -I\
          <...>\AppData\Local\Android\sdk\platforms\android-27\android.jar\
          --manifest\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\merged_manifests\debug\processDebugManifest\merged\AndroidManifest.xml\
          -o\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\processed_res\debug\processDebugResources\out\resources-debug.ap_\
          -R\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\incremental\processDebugResources\resources-list-for-resources-debug.ap_.txt\
          --auto-add-overlay\
          --java\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\generated\not_namespaced_r_class_sources\debug\processDebugResources\r\
          --custom-package\
          com.example.ntcdapp\
          -0\
          apk\
          --output-text-symbols\
          <...>\Development\NTCD.AppMobile\ntcdapp\build\app\intermediates\symbols\debug\R.txt\
          --no-version-vectors
  Daemon:  AAPT2 aapt2-3.2.1-4818971-windows Daemon #0

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
*******************************************************************************************
The Gradle failure may have been because of AndroidX incompatibilities in this Flutter app.
See https://goo.gl/CP92wY for more information on the problem and how to fix it.
*******************************************************************************************
Gradle task assembleDebug failed with exit code 1
```

**This is my android/app/build.gradle**

```
def localProperties = new Properties()
def localPropertiesFile = rootProject.file('local.properties')
if (localPropertiesFile.exists()) {
    localPropertiesFile.withReader('UTF-8') { reader ->
        localProperties.load(reader)
    }
}

def flutterRoot = localProperties.getProperty('flutter.sdk')
if (flutterRoot == null) {
    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
}

def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
if (flutterVersionCode == null) {
    flutterVersionCode = '1'
}

def flutterVersionName = localProperties.getProperty('flutter.versionName')
if (flutterVersionName == null) {
    flutterVersionName = '1.0'
}

apply plugin: 'com.android.application'
apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"

android {
    compileSdkVersion 27

    lintOptions {
        disable 'InvalidPackage'
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId "com.example.ntcdapp"
        minSdkVersion 16
        targetSdkVersion 27
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
        }
    }
}

flutter {
    source '../..'
}

dependencies {
    testImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test:runner:1.0.2'
    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
}
```

**gradle/build.gradle**
```

buildscript {
    repositories {
        google()
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:3.2.1'
    }
}

allprojects {
    repositories {
        google()
        jcenter()
    }
}

rootProject.buildDir = '../build'
subprojects {
    project.buildDir = "${rootProject.buildDir}/${project.name}"
}
subprojects {
    project.evaluationDependsOn(':app')
}

task clean(type: Delete) {
    delete rootProject.buildDir
}
```
 **And my pubspec.yaml**

```
name: ntcdapp
description: A new Flutter project.

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# Read more about versioning at semver.org.
version: 1.0.0+1

environment:
  sdk: ">=2.0.0-dev.68.0 <3.0.0"

dependencies:
  biometric: ^0.0.9
  shared_preferences: ^0.5.2
  font_awesome_flutter: ^8.4.0
  flutter:
    sdk: flutter
  flutter_localizations:
    sdk: flutter
  intl:
  intl_translation:

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.2
  scoped_model: ^0.3.0
  http: ^0.11.3+16
  json_serializable: ^1.5.1
  sqflite: ^1.0.0
  path_provider: ^1.1.0
  connectivity: ^0.4.3+2
  better_uuid: ^0.1.2
  flutter_local_notifications: ^0.7.1

dev_dependencies:
  flutter_test:
    sdk: flutter 
transformers:
- intl_translation:
```

I have tried with the last version (0.7.1+3) and a previous one (0.7.1), but i get the same error. IN my manifest i have included the permissions and receivers described in the documentation

Have any of you went into the same error? 

Thanks a lot!

Luis

**Comment by MaikuB** [comment]
There's a link in the error message that you refer to around AndroidX

**Comment by edisonywh** [comment]
I am also facing the same problem.

I understand that it's probably resolvable by upgrading to AndroidX, but since this issue happens only after implementing this plugin, was wondering if there's a version that does not need AndroidX? (according to the link that you mentioned, Google themselves provide a list of plugins that you should pin to too)

**Comment by MaikuB** [comment]
There's a note in the readme on when this plugin moved to using AndroidX (0.5.0) so you can pick a version before that. There's also a changelog. I would advise to move to AndroidX though, Google's own plugins have done so and most plugin maintainers have been trying to follow suit. If you pick an earlier version then you end up missing fixes and additions that have been added

**Comment by luisms89** [comment]
I solved the problem but I am not sure how and why. I created a new project and I started to add all the dependencies, one by one, to see which one would break the app. Surprisingly, none of them did and the app was running fine with all the same packages I had in my original project. So I copy/pase the build.gradle, pubspec.yaml and the gradle.properties from the new project to my original one (which, by the way, were the same as far as I saw) and suddenly, it worked.
